### PR TITLE
[WIP] multiarch: Pull tarballs

### PR DESCRIPTION
**Pull request by Tim15**
TODO:

- [ ] finish + test `build()` function
- [ ] rewrite `library()` function
- [ ] clean up code

### CODE REVIEW

**Comment by tianon** [comment]
This needs to also do something similar to https://github.com/tianon/docker-brew-alpine-multiarch/blob/d6cc965a58c2b4c7f5ea3c41d078e8625c89fcb6/3.6/x86_64/Dockerfile#L4-L5 for setting the timezone to `UTC` from the default of `GMT`, right?

```dockerfile
# ensure UTC instead of the default GMT
RUN [ ! -e /etc/localtime ] \
	&& apk add --no-cache --virtual .tz-utc tzdata \
	&& cp -vL /usr/share/zoneinfo/UTC /etc/localtime \
	&& apk del .tz-utc
```

**Comment by Tim15** [comment]
Yeah, I'll add that to the new Dockerfiles.

File: build [hunk 1]
```diff
@@ -99,18 +99,57 @@ push() {
...
+			case "$arch" in # converts Alpine Linux arch strings to Docker arch strings
+				x86_64) echo -n "amd64";;
+				x86) echo -n "i386";;
+				armhf) echo -n "arm32v7";; # TODO: What about arm32v7
+				aarch64) echo -n "arm64v8";;
+				ppc64le) echo -n "ppc64le";;
+				s390x) echo -n "s390x";;
+				*) return 1;; # Fail on unknown archs
+			esac
+			if [[ i -ne $#-1 ]]; then
```
Comment (ncopa) [inline build]: is there a missing `$` for `i`?

````sh
  if [[ $i -ne $#-1 ]]; then
````

**Comment by tianon** [comment]
Hey @Tim15, another friendly ping (this is definitely something really important blocking a lot of other work :sweat_smile:) -- anything we can do to help out here?  Is your checklist still accurate for what's left to finish, or is this close to the finish line? :pray:

**Comment by andyshinn** [comment]
I will pick up the rest of this next week.

**Comment by vielmetti** [comment]
A friendly query to @Tim15 @andyshinn regarding status. cc @tianon .

**Comment by andyshinn** [comment]
I'm working on it in another branch. I've simplified a lot of this. But I've run into an issue with the timezones (can't install non-x86 tzdata). If I can't figure out a more elegant solution by tomorrow evening, I'll just copy the tzdata next to the `Dockerfile` and then `COPY` it into the resulting image.

**Comment by vielmetti** [comment]
Thanks @andyshinn . https://wiki.alpinelinux.org/wiki/Setting_the_timezone gives directions to `apk add tzdata`, is that command failing in some way?

**Comment by vielmetti** [comment]
Digging a little more, it looks like #310 was a recent discussion of TZ and tzdata handling.

**Comment by ncopa** [comment]
@andyshinn possible alternatives:
````sh
apk fetch --quiet --stdout tzdata | tar -zx usr/share/zoneinfo && cp usr/share/zoneinfo/UTC /etc/localtime && rm -r usr/share/zoneinfo
````
(apparently the /usr/share/zoneinfo/UTZ is a hardlink so we cannot just extract that file)

other alternative is to embed the encoded UTC data in shell format:
````sh
printf "\x54\x5a\x69\x66\x32\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x55\x54\x43\x00\x00\x00\x54\x5a\x69\x66\x32\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x04\xf8\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x55\x54\x43\x00\x00\x00\x0a\x55\x54\x43\x30\x0a" > /etc/localtime
````

**Comment by andyshinn** [comment]
Closing in lieu of #352.

**Pull request closed without merging**